telosnetwork/teloscan
# Notifications and feedback on transactions

# Fixes #106

## Description
This PR includes a packed solution to simplify the task of notifying the user whenever a transaction is signed and sent, succeeded or not, with a simple call to a generic signCustomTransaction function.

## Test scenarios
- go to this link
- login
- go to Liquid Staking
  - stake, unstake and withdraw
  - in all cases, it should:
    - put the button in the loading state (and stop when done)
    - after confirming in the wallet, a neutral notification should say what is happening (auto dismissed)
    - if it succeeds, a green success notification is shown with a "see transaction" button redirecting to the transaction page
    - if any error occurs, a red error notification is shown with a "see details" button which opens a popup with the error JSON 
- go to a custom contract [like stlos contract](https://deploy-preview-466--teloscan-stage.netlify.app/address/0xa9991E4daA44922D00a78B6D986cDf628d46C4DD)
  - pick a function you know you can call (like approve)
  - fill in the form and press the write button
    - put the button in the loading state (and stop when done)
    - after confirming in the wallet, a neutral notification should say 'calling **foo** with **N** params'' (auto dismissed)
    - if it succeeds, a green success notification is shown with a "see transaction" button redirecting to the transaction page
    - if any error occurs, a red error notification is shown with a "see details" button which opens a popup with the error JSON 

## Screen captures

[Notifications-on-Teloscan.webm](https://github.com/telosnetwork/teloscan/assets/4420760/9ba2dfec-a07c-49c2-b8d3-bc4fcbb07cb2)




## Checklist:
